filter step 0.1

0xChijioke opened the pull request:
## Description

Changes Made:

Added a button on the main view.

When the button is clicked, it opens a modal.
![image](https://github.com/BuidlGuidl/impact-calculator/assets/53488449/731554bb-8235-4d5c-8a8f-73ced420d4fb)

    
 ##   Additional Info

I wasn't entirely sure if the keycap beside the "Set Filters" text on the button was intended to be the keycap emoji or if it was meant to be dynamic. Therefore, I omitted it for now. If it should be the keycap zero emoji, please let me know, and I'll gladly add it.

I've started working on getting an input field at the top of the modal (assumed we might use a reusable input search, so the search is the same as the one waiting to be used for searching vectors but refactored it to take a "placeholder" arg. if it doesn't make sense I can make a dedicated search component for filters and presets)

## Related issue
#79

escottalexander reviewed (changes requested):
A few comments to check over. Thanks for your brain power on this one.

0xChijioke commented on packages/nextjs/components/impact-vector/ImpactVectorTable.tsx:
```diff
@@ -47,6 +74,30 @@ const ImpactVectorTable = () => {
+          <div className="modal-box min-h-[300px] relative">
+            <p className="whitespace-nowrap flex gap-4 text-sm opacity-60 top-0 absolute left-3 items-center">
+              <FiSearch className="w-5 h-5" />
+              Press &apos;/&apos; to search filters and &apos;@&apos; to search presets
```
My bad.

0xChijioke commented:
@escottalexander I've addressed the review comments and made the necessary changes:

    Updated the filter badge to use daisy-ui badge.
    Moved the modal to its component.
    Corrected the placement of the search bar.
    Users can use 'esc' or click outside the dialog to close it.

    Adjusted the positioning of the button to align with the 'impact vector' and 'weight' headers.
    
![image](https://github.com/BuidlGuidl/impact-calculator/assets/53488449/5370e9a3-ed4f-4a19-b130-51113cfaf1d9)

![image](https://github.com/BuidlGuidl/impact-calculator/assets/53488449/6f3d32b8-a9bf-4d3c-8306-8d9a54d7bffc)

    

Let me know what you think!

escottalexander reviewed (approved):
This looks great. Nice job.
